plugins/transformations/json/srcC/FlatTypeTransformation.c

```diff
@@ -1072,25 +507,25 @@ DDS_ReturnCode_t RTI_TSFM_Json_FlatTypeTransformation_serialize(
         if (failed_serialization || self->state->json_buffer == NULL
             || self->state->json_buffer_size == 0) {
             failed_serialization = DDS_BOOLEAN_FALSE;
-            if (DDS_RETCODE_OK
-                != RTI_TSFM_realloc_buffer(
-                        self->config->serialized_size_min,
-                        self->config->serialized_size_incr,
-                        self->config->serialized_size_max,
-                        &self->state->json_buffer,
-                        &self->state->json_buffer_size)) {
-                /* TODO Log error */
+            retcode = RTI_TSFM_realloc_buffer(
+                    self->config->serialized_size_min,
+                    self->config->serialized_size_incr,
+                    self->config->serialized_size_max,
+                    &self->state->json_buffer,
+                    &self->state->json_buffer_size);
+            if (retcode != DDS_RETCODE_OK) {
+                RTI_TSFM_ERROR("unable to realloc_buffer");
```

Comment:
The log macros were defined so that a `;` should not be added when they are used. This allows the macros to result in completely empty lines when they are compiled out, instead of leaving a line with just a semi-colon which used to trip up some compiler (Windows, IIRC).

Reply:
I didn't see they were finished with `;`, however that is not recommended (see coding style https://wiki.sei.cmu.edu/confluence/display/c/PRE11-C.+Do+not+conclude+macro+definitions+with+a+semicolon )

Another example to the ones in that link:
```
retcode != RETCODE_OK ?   RTI_TSFM_ERROR("unable to realloc_buffer") : RTI_TSFM_TRACE("realloc_buffer correctly");
```

Not finishing with `;` also makes autoformatters to not work correctly, that is not a functional issue, but it also helps.

Reply:
Anyway, I'll modify the logging infrastructure as part of #8

Reply:
The reasons for not adding a `;` mentioned by that article are not relevant here, and this is intentional. Micro also uses this pattern. Also, those are not our coding standards. Please be consistent with the rest of the code. And if you end up changing the logging infrastructure, don't get rid of this feature.

Reply:
See the definition of the logging macros here to verify that they all have a `;`: https://github.com/rticommunity/rticonnextdds-gateway/blob/master/common/transformation/srcC/TransformationLog.h#L52